draft-ietf-intarea-proxy-config.md

```diff
@@ -233,38 +233,58 @@ content-length = 222
 The client would learn the URI template of the proxy that supports UDP using {{CONNECT-UDP}},
 at "https://proxy.example.org/masque{?target_host,target_port}".
 
-# Split DNS information for proxies {#split-dns}
+# Destination information for proxies {#destinations}
 
-Split DNS configurations are cases where only a subset of domains is routed through
-a VPN tunnel or a proxy. For example, IKEv2 defines split DNS configuration in
-{{?IKEV2SPLIT=RFC8598}}.
+Destination configuration is used when only a subset of destinations is reachable through
+a proxy. Destination restrictions are often used in VPN tunnel configurations such as split
+DNS in IKEv2 {{?IKEV2SPLIT=RFC8598}}.
 
-PvD Additional Information can be used to indicate that a proxy PvD has a split DNS
-configuration.
+PvD Additional Information can be used to indicate that a proxy PvD should be used for a limited
+set of destinations.
 
 This document defines two optional keys that for subdictionaries in the `proxies`
 array that are used for split-DNS configuration.
 
 | JSON Key | Optional | Description | Type | Example |
 | --- | --- | --- | --- | --- |
-| matchDomains | Yes | An array of DNS zones or subdomains that can be accessed over this proxy | Array of Strings | [ "example.com" ] |
-| excludedDomains | Yes | An array of DNS zones or subdomains that cannot be accessed over this proxy, which should be more specific domains of entries in the matchDomains array | Array of Strings | [ "public.example.com" ] |
+| matchDomains | Yes | An array of hostnames and wildcard FQDNs that can be accessed over this proxy | Array of Strings | [ "www.example.com", "*.local" ] |
+| excludedDomains | Yes | An array of hostnames and wildcard FQDNs that cannot be accessed over this proxy, which should be more specific domains of entries in the matchDomains array | Array of Strings | [ "exclude.local" ] |
+| matchIPs | Yes | An array of IP addresses and subnets that can be accessed over this proxy | Array of Strings | [ "2001:DB8::1", "192.168.1.0/24" ] |
+| excludeIPs | Yes | An array of IP addresses and subnets that cannot be accessed over this proxy, which should be more specific addresses or subnets of entries in the matchIP array | Array of Strings | [ "192.168.1.0/25", "192.168.1.254" ] |
+| matchPorts | Yes | An array of TCP or UDP port ranges accessible over this proxy | Array of Strings | [ "80", "443", "1024-65535" ] |
+| excludePorts | Yes | An array of TCP or UDP port ranges that cannot be accessed over this proxy, which should be more specific than entries in the matchPorts array | Array of Strings | [ "3389", "8443" ] |
```

Comment:
I am not sure about excludePorts as define here. Ports are a limited space.  Therefore you can define everything by only using matchPorts. The parsing will be easier as well.
Maybe excludedPorts should be use only without matchPorts, to define ports that are disallowed, i.e. use one or the other.
We should also define that ports above 65535 is an error, do we want to reject the configuration or just ignore the mistake?

I think we should have a limit of the number of elements in the arrays, or a recommendation that the arrays should not be longer than x elements because the implementation may reject it. Also we should define the behavior in this case, e.g. the config is rejected as a whole or elements after element x are ignored. This is a separate issue though, and I will open a new issue for it.

Reply:
Great ideas!
I'll update PR accordingly.

I don't think there is a concrete list size recommendation that will apply reasonably well to all kinds of clients and use cases - but I think it's a good suggestion to put more important entries in the beginning of the list.

Reply:
Short correction: by "the parsing will be easier as well" I meant to say that it will be easier to create a single list instead of going through 2 lists to parse it into a single list or always check 2 lists.

Reply:
Absolutely! PR amended.